src/xrpld/app/tx/detail/LoanSet.cpp

```diff
@@ -0,0 +1,675 @@
+        JLOG(ctx.j.warn()) << "LoanBroker does not exist.";
+        return tecNO_ENTRY;
+    }
+    auto const brokerOwner = brokerSle->at(sfOwner);
+    auto const counterparty = tx[~sfCounterparty].value_or(brokerOwner);
+    if (account != brokerOwner && counterparty != brokerOwner)
+    {
+        JLOG(ctx.j.warn()) << "Neither Account nor Counterparty are the owner "
+                              "of the LoanBroker.";
+        return tecNO_PERMISSION;
+    }
+
+    auto const borrower = counterparty == brokerOwner ? account : counterparty;
+    if (auto const borrowerSle = ctx.view.read(keylet::account(borrower));
+        !borrowerSle)
+    {
+        // It may not be possible to hit this case, because it'll fail the
+        // signature check with terNO_ACCOUNT.
+        JLOG(ctx.j.warn()) << "Borrower does not exist.";
+        return terNO_ACCOUNT;
+    }
+
+    auto const vault = ctx.view.read(keylet::vault(brokerSle->at(sfVaultID)));
+    if (!vault)
+        // Should be impossible
+        return tefBAD_LEDGER;  // LCOV_EXCL_LINE
+    Asset const asset = vault->at(sfAsset);
+
+    auto const vaultPseudo = vault->at(sfAccount);
+
+    // Check that relevant values can be represented as the vault asset type.
+    // This check is almost duplicated in doApply, but that check is done after
+    // the overall loan scale is known. This is mostly only relevant for
+    // integral (non-IOU) types
+    {
+        for (auto const& field : getValueFields())
+        {
+            if (auto const value = tx[field];
+                value && STAmount{asset, *value} != *value)
+            {
+                JLOG(ctx.j.warn()) << field.f->getName() << " (" << *value
+                                   << ") can not be represented as a(n) "
+                                   << to_string(asset) << ".";
+                return tecPRECISION_LOSS;
+            }
+        }
+    }
+
+    if (auto const ter = canAddHolding(ctx.view, asset))
+        return ter;
+
+    // vaultPseudo is going to send funds, so it can't be frozen.
+    if (auto const ret = checkFrozen(ctx.view, vaultPseudo, asset))
+    {
+        JLOG(ctx.j.warn()) << "Vault pseudo-account is frozen.";
+        return ret;
+    }
+    // borrower is eventually going to have to pay back the loan, so it can't be
+    // frozen now. It is also going to receive funds, so it can't be deep
+    // frozen, but being frozen is a prerequisite for being deep frozen, so
+    // checking the one is sufficient.
+    if (auto const ret = checkFrozen(ctx.view, borrower, asset))
+    {
+        JLOG(ctx.j.warn()) << "Borrower account is frozen.";
+        return ret;
+    }
+    // brokerOwner is going to receive funds if there's an origination fee, so
+    // it can't be deep frozen
+    if (auto const ret = checkDeepFrozen(ctx.view, brokerOwner, asset))
+    {
+        JLOG(ctx.j.warn()) << "Broker owner account is frozen.";
+        return ret;
+    }
+
```

Comment:
I don't see this required per specs check:

The RippleState between the LoanBroker.Account and the Issuer has the lsfLowDeepFreeze or lsfHighDeepFreeze flag set. (The Loan Broker pseudo-account is frozen).

Reply:
> I don't see this required per specs check:
> 
> The RippleState between the LoanBroker.Account and the Issuer has the lsfLowDeepFreeze or lsfHighDeepFreeze flag set. (The Loan Broker pseudo-account is frozen).

Fixed in a3b2ce91defb87b6e870562b27f5b8a37561217c